translations/translator.py

```diff
@@ -1,22 +1,41 @@
-#!/usr/bin/env python3
 import argparse
+import importlib
 import os
 import sys
 import re
 import yaml
 import json
 
+from abc import ABC, abstractmethod
+
+from translations.utils import Utilities, ConfigUtilities
+
 program = os.path.basename(sys.argv[0])
 config_file = 'translation-config.yml'
 
+
 class Driver:
+    whoami = __qualname__
+
     def main(self, args=sys.argv[1:], prog=program):
         options = self.parse_args(args, prog)
-        data = self.load_config()
-        Validator().validate(data)
-        all_bundles = Bundler().gather(data)
-        if options.generate:
-            TranslationGenerator(options, all_bundles).generate_all()
+        config = self.load_config()
+        Validator().validate(config)
+        exporter = self.instantiate_exporter(config, options)
+        importer = self.instantiate_importer(config, options)
+        all_bundles = Bundler().gather(config)
+
+        manifest = TranslationGenerator(options, all_bundles).generate()
+        if options.view:
+            Utilities.print_data(manifest.data)
+        if options.export:
+            exporter.generate_request(manifest)
+        if options.import_translations:
+            translation_updates, new_messages = importer.process_response(manifest)
+            TranslationUpdater.update(translation_updates)
+            SnapshotUpdater(config).update(manifest, new_messages)
+            Utilities.write_to_json_file('translations-manifest', manifest.data)
+            Utilities.write_to_json_file('translations-updates', translation_updates)
```

Comment:
I think we should separate out everything related to `importing` and `exporting` from the core script out into another script. My initial goal for this script was to simply be able to generate a manifest that you should be able to save out into a file and consume elsewhere. If you consider the manifest to be a file, then you can truly decouple this from the rest of the workflow and plug in whatever you want to do whatever you want with the manifest

Reply:
I agree that the script is becoming too big, and can be modulated better, but I don't see the benefit of saving the manifest. For an import or an export, that manifest should be created first and represent the current state, so why save it and have to execute two (or more scripts)?

Reply:
I was thinking of the use case where someone was planning to use a medium like Amazon, NLTK or Google for their translation services, you would deal mostly with straight json

Btw, I'm fine with all these changes for our use case. As long as it works for ICC and Appian, I don't particularly care. I worked on this to make it a lot more generic for developers to use in repos. And maybe myself in personal projects 😄 . Maybe we can talk about simply keeping an Appian version of this tool in our repos somewhere? Then we will all be on the same page that this works for our use case